Packs/AWS-Athena/Integrations/AWS-Athena/AWS-Athena.py

```diff
@@ -44,205 +19,200 @@ def default(self, obj):
-    client = aws_session(
-        region=args.get('region'),
-        roleArn=args.get('roleArn'),
-        roleSessionName=args.get('roleSessionName'),
-        roleSessionDuration=args.get('roleSessionDuration'),
+    }
+
+    return CommandResults(
+        outputs_prefix='AWS.Athena.Query',
+        outputs_key_field='QueryExecutionId',
+        outputs=context_data,
+        raw_response=response,
+        readable_output=tableToMarkdown('AWS Athena Query Start', context_data),
     )
 
-    response = client.stop_query_execution(QueryExecutionId=args.get('QueryExecutionId'))
-    if response['ResponseMetadata']['HTTPStatusCode'] == 200:
-        demisto.results("The Query {query} was Deleted ".format(query=args.get('QueryExecutionId')))
 
+def stop_query_command(args: dict, client):
+    query_execution_id: str = args['QueryExecutionId']
+    response = client.stop_query_execution(QueryExecutionId=query_execution_id)
+
+    if response.get('ResponseMetadata', {}).get('HTTPStatusCode') == 200:
+        return CommandResults(readable_output=f"Query '{query_execution_id}' has been successfully stopped.")
+
+    else:
+        demisto.debug("Response:\n" + str(response))
+        raise DemistoException(f"Failed to stop query '{query_execution_id}'.")
+
+
+@polling_function(
+    name=demisto.command(),
+    interval=arg_to_number(demisto.args().get('interval_in_seconds', 10)),
+    timeout=arg_to_number(demisto.args().get('timeout_in_seconds', 300)),
+    requires_polling_arg=True,
+)
+def get_query_execution_command(args: dict, client):
+    query_execution_id: str = args['QueryExecutionId']
+    raw_response = client.get_query_execution(QueryExecutionId=query_execution_id)
 
-def get_query_execution_command(args):
-    client = aws_session(
-        region=args.get('region'),
-        roleArn=args.get('roleArn'),
-        roleSessionName=args.get('roleSessionName'),
-        roleSessionDuration=args.get('roleSessionDuration'),
-    )
-    kwargs = {'QueryExecutionId': args.get('QueryExecutionId')}
-    response = client.get_query_execution(**kwargs)
     try:
-        raw = json.loads(json.dumps(response, cls=DatetimeEncoder))
+        raw_response = json.loads(json.dumps(raw_response, cls=DatetimeEncoder))
+
     except ValueError as e:
-        return_error('Could not decode/encode the raw response - {err_msg}'.format(err_msg=e))
-    ec = {'AWS.Athena.Query(val.QueryExecutionId === obj.QueryExecutionId)': raw}
-    human_readable = tableToMarkdown('AWS Athena Query', raw)
-    return_outputs(human_readable, ec)
-
-
-def get_query_results_command(args):
-    client = aws_session(
-        region=args.get('region'),
-        roleArn=args.get('roleArn'),
-        roleSessionName=args.get('roleSessionName'),
-        roleSessionDuration=args.get('roleSessionDuration'),
+        return_error('Could not parse the received response.')
+        demisto.debug(f'Error:\n{e}\n'
+                      f'Response:\n{raw_response}')
```

Comment:
The line after `return_error` is unreachable since it exits.
Please fix it and raise an exception instead of calling return_error.